pkg/vault/instances.go

```diff
@@ -231,6 +243,37 @@ func configureKubeAuthClient(client *api.Client, bundle AuthBundle) error {
 	return nil
 }
 
+func configureAppRoleAuthClient(client *api.Client, roleID, secretID string) (error, *api.Secret) {
+	auth, err := approle.NewAppRoleAuth(
+		roleID,
+		&approle.SecretID{FromString: secretID},
+	)
+	if err != nil {
+		return err, nil
+	}
+
+	var secret *api.Secret
+	err = utils.Retry(defaultTokenRetryAttempts, defaultTokenRetrySleep, func() error {
+		s, err := client.Auth().Login(context.TODO(), auth)
+		if err != nil {
+			// Writing to the AppRole mount endpoint to "login" to obtain a new
+			// token might return an empty response without necessarily failing.
+			if strings.Contains(err.Error(), "client token not set") {
```

Comment:
**To the reviewer:**

This is very unfortunate. However, Vault tends to return bare errors without using custom types; thus, there is little one can do here to differentiate between specific error types.